rideshare_nguyen.rb

```diff
@@ -0,0 +1,153 @@
+########################################################
+# Diana Nguyen - Space
+# Ride Share
+########################################################
+# (0) Setting up the entire data structure:
+# Level 1: master hash "rides"
+# Level 2: 4 sub hashes for each of the drivers
+#         driver id is the key and the values are different
+#         variables (e.g. date of ride they gave)
+# Level 3: within the values hold the array for the
+#         variables, this is so the variables line up,
+#         (e.g. for Driver 1, they had a date of ride of
+#          Feb 3rd, with a rating of 3, earning $10, and 
+#          the rider had an id of RD0003)
+
+rides = {
+  :DR0001 => {
+  date_of_ride: ["3rd Feb 2016", "3rd Feb 2016", "5th Feb 2016"],
+  rating: [3, 4, 2],
+  money_earned: [10, 30, 45],
+  rider_of_drive: ["RD0003", "RD0015", "RD0003"]
+  },
+  :DR0002 => {
+  date_of_ride: ["3rd Feb 2016", "4th Feb 2016", "5th Feb 2016"],
+  rating: [5, 1, 3],
+  money_earned: [25, 15, 35],
+  rider_of_drive: ["RD0073", "RD0013", "RD0066"]
+  },
+  :DR0003 => {
+  date_of_ride: ["4th Feb 2016", "5th Feb 2016"],
+  rating: [5, 2],
+  money_earned: [5, 50],
+  rider_of_drive: ["RD0066", "RD0003"]
+  },
+  :DR0004 => {
+  date_of_ride: ["3rd Feb 2016", "4th Feb 2016", "5th Feb 2016"],
+  rating: [5, 4, 5],
+  money_earned: [5, 10, 20],
+  rider_of_drive: ["RD0022", "RD0022", "RD0073"]
+  }
+}
+
+# Welcome statement of data report
+
+puts "Here is the Ride Share Data analysis from February 3rd, 2016 to February 5th, 2016: "
+puts "\n"
+
+########################################################
+# (1) Number of rides each driver has given
+# Based off of the "date_of_rides" array, finding the 
+# length of this array will give the total amount of rides
+
+# Strategy - first try to find amount for one driver first, 
+# then iterate. 
+# Came up with the following code for a single driver
+## puts "Driver 1 gave #{(rides[:DR0001][:date_of_ride]).length} rides"
+
+# To iterate the above code to give all drivers amount of rides
+# use the ".map" enumerable
+# to access the correct array, must include both key (driver) and value (data) 
+# as arguments for the block
```

Comment:
this is pretty big for a standard comment! Can you sum it up in 1-2 lines instead? Comments work best as an in-line tool, snug with the code it's giving info on.